.github/workflows/rust.yml

```diff
@@ -84,22 +84,49 @@ jobs:
         components: rust-src
     - name: Install deps
       run: |
-        Start-BitsTransfer -Source https://ftp.mozilla.org/pub/mozilla/libraries/win32/MozillaBuildSetup-3.4.exe -Destination ./MozillaBuildSetup.exe
+        Start-BitsTransfer -Source https://ftp.mozilla.org/pub/mozilla/libraries/win32/MozillaBuildSetup-4.0.2.exe -Destination ./MozillaBuildSetup.exe
         .\MozillaBuildSetup.exe /S | Out-Null
-        echo 'C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Tools\LLVM\bin' | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
+        iwr -useb get.scoop.sh -outfile 'install.ps1'
+        .\install.ps1 -RunAsAdmin
+        scoop install llvm@15.0.7 --global
+        echo "C:\ProgramData\scoop\shims;C:\Users\runneradmin\scoop\shims" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
     - name: Run sccache-cache
       uses: mozilla-actions/sccache-action@v0.0.3
       with:
         version: ${{ env.SCCACHE_VERSION }}
+    # TODO: Remove this step and start-shell.bat when the MozillaBuild is updated
+    - name: Replace MozillaBuild shell
+      run: |
+        Remove-Item -Path C:\mozilla-build\start-shell.bat
+        Move-Item -Path start-shell.bat -Destination C:\mozilla-build\
+    - uses: dtolnay/rust-toolchain@stable
+      if: (!contains(matrix.target, 'uwp'))
+    - uses: dtolnay/rust-toolchain@master
+      if: contains(matrix.target, 'uwp')
+      with:
+        toolchain: "nightly-2023-02-02"
+        components: rust-src
     - name: Build uwp
       if: contains(matrix.target, 'uwp')
-      shell: cmd
+      env:
```

Comment:
This was also refactored in https://github.com/servo/mozjs/commit/16c72d786f623f6844c315c7fe8a9fb19d713f13

Reply:
I've also updated this as well.